### PR TITLE
WotW - Visiting Advisor

### LINKED ISSUE #2957: WotW: Visiting Advisor

### PR DESCRIPTION
**Pull request by jamesoddy**
Closes #2957

### CODE REVIEW

**Comment by jeremylarner** [comment]
Does this work if Visiting Advisor is the only character in the conflict?

**Comment by jamesoddy** [comment]
It did work if Visiting Advisor was your only participating character (you just have to click the prompt for "No more targets").  I've added a test for that situation.

I have updated as you suggested though - my only concern with that was that the actions were not "combined" and there could be some weird reaction interaction.  But there is nothing that I can think of off the top of my head that would cause an issue.

**Comment by jeremylarner** [comment]
The tests for this will break when combined with #3013.  The implementation may break too :P

File: server/game/cards/07-WotW/VisitingAdvisor.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            target: {
+                controller: Players.Self,
+                cardType: CardTypes.Character,
+                optional: true,
+                cardCondition: (card, context) => card !== context.source,
+                gameAction: AbilityDsl.actions.sendHome()
+            },
+            gameAction: AbilityDsl.actions.sendHome(context => ({ target: context.source })),
+            effect: 'send {0}{1}{2} home',
+            effectArgs: (context) => context.target.length === 0 ? [context.source] : [' and ', context.source]
```
Comment (jeremylarner) [inline server/game/cards/07-WotW/VisitingAdvisor.js]: This looks awkward to me.  Do you need an `effect` at all here?  Doesn't the default just work(TM)?

Comment (jamesoddy) [reply server/game/cards/07-WotW/VisitingAdvisor.js]: Damn right it was awkward, but as it is also affecting `context.source` (Visiting Advisor itself) this was the only way I could get it to display the right messages.

**Pull request merged**